apollo-entity/src/test/java/com/spotify/apollo/entity/CodecTest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * -\-\-
+ * Spotify Apollo Entity Middleware
+ * --
+ * Copyright (C) 2013 - 2016 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.apollo.entity;
+
+import static com.fasterxml.jackson.databind.PropertyNamingStrategy.CAMEL_CASE_TO_LOWER_CASE_WITH_UNDERSCORES;
+import static com.spotify.apollo.entity.EntityMiddlewareTest.await;
+import static com.spotify.apollo.entity.JsonMatchers.asStr;
+import static com.spotify.apollo.entity.JsonMatchers.hasJsonPath;
+import static com.spotify.apollo.test.unit.ResponseMatchers.hasHeader;
+import static com.spotify.apollo.test.unit.ResponseMatchers.hasPayload;
+import static com.spotify.apollo.test.unit.ResponseMatchers.hasStatus;
+import static com.spotify.apollo.test.unit.StatusTypeMatchers.withCode;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.spotify.apollo.AppInit;
+import com.spotify.apollo.RequestContext;
+import com.spotify.apollo.Response;
+import com.spotify.apollo.Status;
+import com.spotify.apollo.entity.EntityCodecsTest.Entity;
+import com.spotify.apollo.route.Middleware;
+import com.spotify.apollo.route.Route;
+import com.spotify.apollo.test.ServiceHelper;
+import io.norberg.automatter.jackson.AutoMatterModule;
+import okio.ByteString;
+import org.junit.Test;
+
+public class CodecTest {
```

Comment:
This is more like `EntityMiddlewareTest`, I think? Codec tests would validate code behaviours, such as whether they perform content negotiation correctly, etc.

Reply:
Hmm, there's no actual implementation of content negotiation in the codebase right now. I've done a very simple assertion that the `RequestContext` is being passed into the `Codec` though. I assume that should be enough to allow any code to make decisions on it's own based on that.

Reply:
Absolutely, I just meant I don't feel like this class is actually testing codecs, it's testing `EntityMiddleware` ?

Reply:
That's true

Reply:
Renamed test class